packages/block-library/src/heading/edit.js

```diff
@@ -71,43 +59,42 @@ function HeadingEdit( {
 					<p>{ __( 'Level' ) }</p>
 					<HeadingToolbar isCollapsed={ false } minLevel={ 1 } maxLevel={ 7 } selectedLevel={ level } onChange={ ( newLevel ) => setAttributes( { level: newLevel } ) } />
 				</PanelBody>
-				<HeadingColorUI
-					setTextColor={ setTextColor }
-					textColorValue={ textColor.color }
-				/>
 			</InspectorControls>
-			<RichText
-				identifier="content"
-				tagName={ tagName }
-				value={ content }
-				onChange={ ( value ) => setAttributes( { content: value } ) }
-				onMerge={ mergeBlocks }
-				onSplit={ ( value ) => {
-					if ( ! value ) {
-						return createBlock( 'core/paragraph' );
-					}
+			{ InspectorControlsColorPanel }
+			<TextColor>
+				<RichText
+					identifier="content"
+					tagName={ tagName }
+					value={ content }
+					onChange={ ( value ) => setAttributes( { content: value } ) }
+					onMerge={ mergeBlocks }
+					onSplit={ ( value ) => {
+						if ( ! value ) {
+							return createBlock( 'core/paragraph' );
+						}
 
-					return createBlock( 'core/heading', {
-						...attributes,
-						content: value,
-					} );
-				} }
-				onReplace={ onReplace }
-				onRemove={ () => onReplace( [] ) }
-				className={ classnames( className, {
-					[ `has-text-align-${ align }` ]: align,
-					'has-text-color': textColor.color,
-					[ textColor.class ]: textColor.class,
-				} ) }
-				placeholder={ placeholder || __( 'Write heading…' ) }
-				style={ {
-					color: textColor.color,
-				} }
-			/>
+						return createBlock( 'core/heading', {
+							...attributes,
+							content: value,
+						} );
+					} }
+					onReplace={ onReplace }
+					onRemove={ () => onReplace( [] ) }
+					className={ classnames( className, {
+						[ `has-text-align-${ align }` ]: align,
+					} ) }
+					placeholder={ placeholder || __( 'Write heading…' ) }
+				/>
+			</TextColor>
 		</>
 	);
 }
 
-export default compose( [
-	withColors( 'backgroundColor', { textColor: 'color' } ),
-] )( HeadingEdit );
+export default withFallbackStyles( ( node ) => {
```

Comment:
Can you explain Why did we add this new behavior? (or where was it handled before?
also, is this something that can be absorbed by the hook somehow? Is this specific to the heading block.

Reply:
>Can you explain Why did we add this new behavior?

For the background value of the contrast checker.

>or where was it handled before?

The Heading block did not have a contrast checker.

>is this something that can be absorbed by the hook somehow? Is this specific to the heading block.

No it's not specific, it could potentially be useful for any block that changes text colors, but uses a parent background.

I think that if the contrast checker config is provided to `useColors` without a background value, the hook should somehow default to doing this parent traversal, what do you think?

Reply:
> I think that if the contrast checker config is provided to useColors without a background value, the hook should somehow default to doing this parent traversal, what do you think?

This means that it will be automatically handled by the hook right?  I think that's better. I wonder if we need a way to disable it if needed?

Reply:
Yes, what do you think about this proposal?

https://github.com/WordPress/gutenberg/pull/18148#discussion_r340254660

Reply:
See #18237.